Sorry, I'm a bit confused with the CI here - it only seems to test readers?

Failure with [#27915f7](https://github.com/intake/intake/pull/888/commits/27915f7f5ae7c3d43a3eb5a2ea5c0cce1fe31d06):
___
```python
def test_expand_defaults():
    """
    expand_defaults should not warn that getshell=False when it is True.
    """
    with warnings.catch_warnings():
        warnings.filterwarnings("error", category=UserWarning)
        utils.expand_defaults(default="v2000-01-01", client=False, getenv=True, getshell=True)

    assert True
```

raises: `FAILED intake/catalog/tests/test_utils.py::test_expand_defaults - UserWarning: Shell command not executed due to getshell=False`

--- a/intake/catalog/utils.py
+++ b/intake/catalog/utils.py
@@ -169,7 +169,7 @@ def expand_defaults(default, client=False, getenv=True, getshell=False):
             default = subprocess.check_output(cmd).rstrip().decode("utf8")
         except (subprocess.CalledProcessError, OSError):
             default = ""
-    else:
+    elif not getshell:
         warnings.warn("Shell command not executed due to getshell=False")
     r = re.match(r"client_shell\((.*)\)", default)
     if r and client and getshell:
@@ -178,7 +178,7 @@ def expand_defaults(default, client=False, getenv=True, getshell=False):
             default = subprocess.check_output(cmd).rstrip().decode("utf8")
         except (subprocess.CalledProcessError, OSError):
             default = ""
-    else:
+    elif not getshell:
         warnings.warn("Shell command not executed due to getshell=False")
     return default
 

